pkg/console/subresource/route/route.go

```diff
@@ -41,53 +90,41 @@ func GetOrCreate(ctx context.Context, client routeclient.RoutesGetter, required
 }
 
 func DefaultRoute(cr *operatorv1.Console) *routev1.Route {
-	route := Stub()
+	route := DefaultStub()
 	route.Spec = routev1.RouteSpec{
 		To:             toService(),
 		Port:           port(),
-		TLS:            tls(),
+		TLS:            tls(nil),
 		WildcardPolicy: wildcard(),
 	}
 	util.AddOwnerRef(route, util.OwnerRefFrom(cr))
 	return route
 }
 
-func Stub() *routev1.Route {
+func DefaultStub() *routev1.Route {
 	meta := util.SharedMeta()
 	return &routev1.Route{
 		ObjectMeta: meta,
 	}
 }
 
-// we can't just blindly apply the route, we need the route.Spec.Host
-// and we don't want to trigger a sync loop.
-// TODO: evaluate metadata.annotations to see what will affect our route in
-// an undesirable way:
-// - https://docs.openshift.com/container-platform/3.9/architecture/networking/routes.html#alternateBackends
-func Validate(route *routev1.Route) (*routev1.Route, bool) {
-	changed := false
-
-	if toServiceSame := equality.Semantic.DeepEqual(route.Spec.To, toService()); !toServiceSame {
-		changed = true
-		route.Spec.To = toService()
-	}
-
-	if portSame := equality.Semantic.DeepEqual(route.Spec.Port, port()); !portSame {
-		changed = true
-		route.Spec.Port = port()
-	}
-
-	if tlsSame := equality.Semantic.DeepEqual(route.Spec.TLS, tls()); !tlsSame {
-		changed = true
-		route.Spec.TLS = tls()
-	}
-
-	if wildcardSame := equality.Semantic.DeepEqual(route.Spec.WildcardPolicy, wildcard()); !wildcardSame {
-		changed = true
-		route.Spec.WildcardPolicy = wildcard()
+func CustomRoute(cr *operatorv1.Console, tlsConfig *CustomTLSCert) *routev1.Route {
+	route := &routev1.Route{
```

Comment:
Nit:  could probably use the `DefaultStub()` and then just override the `.Name`.